src/main/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/validation/CredentialsValidator.java

```diff
@@ -53,39 +54,44 @@ public boolean validate() {
         final String regex = this.esSettings.get(ConfigConstants.OPENDISTRO_SECURITY_RESTAPI_PASSWORD_VALIDATION_REGEX, null);
 
         if ((request.method() == RestRequest.Method.PUT || request.method() == RestRequest.Method.PATCH)
-                && regex != null
-                && !regex.isEmpty()
                 && this.content != null
                 && this.content.length() > 1) {
             try {
                 final Map<String, Object> contentAsMap = XContentHelper.convertToMap(this.content, false, XContentType.JSON).v2();
-                if (contentAsMap != null && contentAsMap.containsKey("password")) {
-                    final String password = (String) contentAsMap.get("password");
-
-                    // Password can be null/empty for an existing user. Regex will validate password if present
-                    if (password != null && !password.isEmpty() && !regex.isEmpty() && !Pattern.compile("^"+regex+"$").matcher(password).matches()) {
-                        if(log.isDebugEnabled()) {
-                            log.debug("Regex does not match password");
-                        }
+                String password = (String) contentAsMap.get("password");
```

Comment:
nit: use final same as original `final String password = (String) contentAsMap.get("password");`

Reply:
Can `contentAsMap` be null? I see the null check is removed

Reply:
`final Map<String, Object> contentAsMap = XContentHelper.convertToMap(this.content, false, XContentType.JSON).v2();` prevents null `contentAsMap`.

Reply:
Does `prevents null` mean it throws null pointer exception or returns empty map?